### ISSUE TITLE
loose route transport enforce breaks some devices

### ISSUE THREAD

**Issue by etamme** [OpenSIPS/opensips#957]
This commit

https://github.com/OpenSIPS/opensips/commit/f91441d43e814d06ebf325e637206be411112879#diff-1b32938867955c53e944e1cf69a522eaL231

Breaks loose routing for polycom phones in the case where the UAC does not explicity specify a transport parameter.  The polycom will insert a transport=invalid param on a sequential request causing the loose route to fail.

See an example sip trace here: https://gist.github.com/etamme/8943fcbdade672b38f780a6742241f88#file-gistfile1-txt-L119

A proposed fix would be to do some thing like:

`diff --git a/modules/rr/loose.c b/modules/rr/loose.c
index a8e5a1f..23bcae5 100644
--- a/modules/rr/loose.c
+++ b/modules/rr/loose.c
@@ -232,6 +232,10 @@ static inline int is_myself(struct sip_uri* _uri)
 
        port = get_uri_port(_uri, &proto);
 
+       if (proto == PROTO_OTHER) {
+         proto = PROTO_NONE;
+       }
+
        ret = check_self(&_uri->host, port, proto);
        if (ret < 0) return 0;
 
`

**Comment by bogdan-iancu** [comment]
@etamme , @james-criscuolo , what is the idea of Polycom adding the "transport=invalid" when mirroring the received RR header ? according to RFC3261 is a bit illegal (should do a perfect mirroring) and makes no sense (the invalid value is not defined in RFC with a certain meaning).

**Comment by etamme** [comment]
I agree - I think that the polycom is not doing the right thing here, and this is a bandaid to work around the problem.  I will open a issue with polycom, and try to find out how wide spread the issue is - e.g. is  it on every VVX model on all firmwares?

**Comment by pvint** [comment]
@etamme   We just came across this same issue. Through some experimentation we have discovered that with Polycom VVX phones with firmare <=5.3.x this does not occur, but with firmwares >=5.4.x we get the transport=invalid from the Polycom on the BYE packet (and then bad things happen....)

**Comment by pvint** [comment]
I just wanted to add a couple findings on this issue:
This only seems to occur when the Polycom is provisioned with:
`voIpProt.server.1.transport = "DNSnaptr" `

With transport to "UDPonly" (or other protocols) eliminates the issue.

**Comment by bogdan-iancu** [comment]
Resuming here : rather than forcing the conversion of PROTO_OTHER  to PROTO_NONE (which is a bit pron to errors), I would better add the `transport` parameter all the time  (even for `UDP`) when doing `record_route()` - I guess this will make Polycom happy, right ?
